File: Angular/src/app/models/user-conversation-messages-summary.ts
import { JSONObject } from "ts-json-object"

export class UserConversationMessagesSummary extends JSONObject {
  email!: string;
  fullname!: string;
  title!: string;
  number_messages!: number;
  length_messages!: number;
  avg_length_messages!: number;
  conversation_created_at!: string;
  number_days_resolution!: number;
  status!: string;
}

File: Rails/app/models/user_conversation_messages_summary.rb
class UserConversationMessagesSummary < ApplicationRecord
    self.table_name = "user_conversation_messages_summary"
    before_validation do
        throw :abort
    end
    attribute :email, :string
    attribute :fullname, :string
    attribute :title, :string 
    attribute :number_messages, :integer
    attribute :length_messages, :integer
    attribute :avg_length_messages, :integer
    attribute :conversation_created_at, :string
    attribute :number_days_resolution, :integer
    attribute :status, :string
end
